File: ployglots/mta-ployglot-pkg-figma-api/packages/ts/src/resources/dev-resources.ts
// @ts-nocheck
/**
 * Dev Resources — Figma REST API coverage.
 *
 * Tier: basic. Dev Mode resources link a node to an external URL
 * (Jira ticket, Storybook page, etc.).
 */

import type { FigmaClient } from '../figma-client.js';

export interface DevResourceCreateItem {
  name: string;
  url: string;
  file_key: string;
  node_id: string;
}

export interface DevResourceUpdateItem {
  id: string;
  name?: string;
  url?: string;
}

export class DevResourcesResource {
  constructor(private readonly client: FigmaClient) {}

  /** GET /v1/files/:file_key/dev_resources — dev resources scoped to a file. */
  async list(fileKey: string, nodeIds?: string[]): Promise<unknown> {
    const params: Record<string, string> = {};
    if (nodeIds && nodeIds.length > 0) params.node_ids = nodeIds.join(',');
    return this.client.get(`/v1/files/${encodeURIComponent(fileKey)}/dev_resources`, {
      params,
    });
  }

  /** POST /v1/dev_resources — bulk-create dev resources. */
  async create(items: DevResourceCreateItem[]): Promise<unknown> {
    return this.client.post('/v1/dev_resources', { body: { dev_resources: items } });
  }

  /** PUT /v1/dev_resources — bulk-update dev resources. */
  async update(items: DevResourceUpdateItem[]): Promise<unknown> {
    return this.client.post('/v1/dev_resources', { body: { dev_resources: items } });
  }

  /** DELETE /v1/files/:file_key/dev_resources/:id — delete one. */
  async delete(fileKey: string, devResourceId: string): Promise<void> {
    await this.client.delete<void>(
      `/v1/files/${encodeURIComponent(fileKey)}/dev_resources/${encodeURIComponent(devResourceId)}`,
    );
  }
}

File: ployglots/mta-ployglot-pkg-figma-api/packages/py/figma_api/resources/dev_resources.py
"""Dev Resources — Figma REST API coverage.

Tier: basic. Dev Mode resources link nodes to external URLs.
"""

from __future__ import annotations

from typing import TYPE_CHECKING, Any
from urllib.parse import quote

if TYPE_CHECKING:
    from .._figma_client import FigmaClient


class DevResourcesResource:
    def __init__(self, client: FigmaClient) -> None:
        self._client = client

    async def list(
        self, file_key: str, node_ids: list[str] | None = None
    ) -> dict[str, Any]:
        """GET /v1/files/:file_key/dev_resources — dev resources scoped to a file."""
        params: dict[str, Any] = {}
        if node_ids:
            params["node_ids"] = ",".join(node_ids)
        return await self._client.get(
            f"/v1/files/{quote(file_key, safe='')}/dev_resources", params=params
        )

    async def create(self, items: list[dict[str, Any]]) -> dict[str, Any]:
        """POST /v1/dev_resources — bulk-create."""
        return await self._client.post("/v1/dev_resources", json={"dev_resources": items})

    async def update(self, items: list[dict[str, Any]]) -> dict[str, Any]:
        """PUT /v1/dev_resources — bulk-update (mapped to POST by the client)."""
        return await self._client.post("/v1/dev_resources", json={"dev_resources": items})

    async def delete(self, file_key: str, dev_resource_id: str) -> None:
        """DELETE /v1/files/:file_key/dev_resources/:id — delete one."""
        await self._client.delete(
            f"/v1/files/{quote(file_key, safe='')}/dev_resources/"
            f"{quote(dev_resource_id, safe='')}"
        )
